### PR TITLE
macOS arm64 support

### PR DESCRIPTION
**Pull request by mihaiplesa**
Resolves 

## Submitter Checklist:

- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: build/rust/config.gni [hunk 1]
```diff
@@ -23,6 +23,9 @@ if (is_win) {
   if (current_cpu == "x64") {
     rustc_target = "x86_64-apple-darwin"
   }
+  else if (current_cpu == "arm64") {
```
Comment (mherrmann) [inline build/rust/config.gni:26]: I would have placed the `else if` on the same line (25) as the preceding curly brace.

**Review by mherrmann — APPROVED** [review]
Looks good to me.

**Review by simonhong — APPROVED** [review]
builds/config.gni looks good to me.

File: components/brave_stats/browser/brave_stats_updater_util.cc [hunk 2]
```diff
@@ -30,7 +30,10 @@ std::string GetPlatformIdentifier() {
   else
     return "winx64-bc";
 #elif defined(OS_MAC)
-  return "osx-bc";
+  if (base::SysInfo::CPUModelName().find("Intel") != std::string::npos)
+    return "osx-bc";
+  else
+    return "osx-arm64-bc";
```
Comment (emerick) [inline components/brave_stats/browser/brave_stats_updater_util.cc:36]: This new platform identifier (`osx-arm64-bc`) will need to be supported on the server as well, from what I understand.

cc: @aekeus

File: components/brave_stats/browser/brave_stats_updater_util.cc [hunk 3]
```diff
@@ -30,7 +30,10 @@ std::string GetPlatformIdentifier() {
   else
     return "winx64-bc";
 #elif defined(OS_MAC)
-  return "osx-bc";
+  if (base::SysInfo::CPUModelName().find("Intel") != std::string::npos)
```
Comment (emerick) [inline components/brave_stats/browser/brave_stats_updater_util.cc:33]: Maybe we could check something like `ARCH_CPU_ARM64` here?

File: components/brave_stats/browser/brave_stats_updater_util.cc (hunk 2 above)
Comment (mihaiplesa) [reply components/brave_stats/browser/brave_stats_updater_util.cc:36]: https://github.com/brave-intl/go-vault/issues/55

File: components/brave_stats/browser/brave_stats_updater_util.cc (hunk 3 above)
Comment (emerick) [reply components/brave_stats/browser/brave_stats_updater_util.cc]: OK, makes sense.

Comment (mihaiplesa) [reply components/brave_stats/browser/brave_stats_updater_util.cc]: tested now and works fine, thanks for the tip

File: build/rust/config.gni (hunk 1 above)
Comment (mihaiplesa) [reply build/rust/config.gni]: done

**Pull request merged**